pkg/cmd/package/support/pack.go

```diff
@@ -0,0 +1,276 @@
+package support
+
+import (
+	"archive/zip"
+	"errors"
+	"fmt"
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/OctopusDeploy/cli/pkg/factory"
+	"github.com/OctopusDeploy/cli/pkg/question"
+	"github.com/OctopusDeploy/cli/pkg/util"
+	"github.com/OctopusDeploy/cli/pkg/util/flag"
+	"github.com/bmatcuk/doublestar/v4"
+	"github.com/spf13/cobra"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+const (
+	FlagId        = "id"
+	FlagVersion   = "version"
+	FlagBasePath  = "base-path"
+	FlagOutFolder = "out-folder"
+	FlagInclude   = "include"
+	FlagVerbose   = "verbose"
+	FlagOverwrite = "overwrite"
+)
+
+type PackageCreateFlags struct {
+	Id        *flag.Flag[string]
+	Version   *flag.Flag[string]
+	BasePath  *flag.Flag[string]
+	OutFolder *flag.Flag[string]
+	Include   *flag.Flag[[]string]
+	Verbose   *flag.Flag[bool]
+	Overwrite *flag.Flag[bool]
+}
+
+type PackageCreateOptions struct {
+	*PackageCreateFlags
+	Writer   io.Writer
+	Ask      question.Asker
+	NoPrompt bool
+	CmdPath  string
+}
+
+func NewPackageCreateFlags() *PackageCreateFlags {
+	return &PackageCreateFlags{
+		Id:        flag.New[string](FlagId, false),
+		Version:   flag.New[string](FlagVersion, false),
+		BasePath:  flag.New[string](FlagBasePath, false),
+		OutFolder: flag.New[string](FlagOutFolder, false),
+		Include:   flag.New[[]string](FlagInclude, false),
+		Verbose:   flag.New[bool](FlagVerbose, false),
+		Overwrite: flag.New[bool](FlagOverwrite, false),
+	}
+}
+
+func NewPackageCreateOptions(f factory.Factory, flags *PackageCreateFlags, cmd *cobra.Command) *PackageCreateOptions {
+	return &PackageCreateOptions{
+		PackageCreateFlags: flags,
+		Writer:             cmd.OutOrStdout(),
+		Ask:                f.Ask,
+		NoPrompt:           !f.IsPromptEnabled(),
+		CmdPath:            cmd.CommandPath(),
+	}
+}
+
+func PackageCreatePromptMissing(opts *PackageCreateOptions) error {
+	if opts.Id.Value == "" {
+		if err := opts.Ask(&survey.Input{
+			Message: "Package ID",
+			Help:    "The ID of the package.",
+		}, &opts.Id.Value, survey.WithValidator(survey.ComposeValidators(
+			survey.Required,
+			survey.MaxLength(200),
+			survey.MinLength(1),
+		))); err != nil {
+			return err
+		}
+	}
+
+	if opts.Version.Value == "" {
+		if err := opts.Ask(&survey.Input{
+			Message: "Version",
+			Help:    "The version of the package; must be a valid SemVer; defaults to a timestamp-based version.",
```

Comment:
could precalculate the default timestamped version and show it in the prompt, this might not be a great idea but something to consider

Reply:
default is actually never set, removed text.